### PR TITLE
Inital changes for custom name

### PR DESCRIPTION
**Pull request by shaunakdixit**
### Contains

It allows You to change name of worlds while creating game.

### How to test

click on advance game setup then 
add world 
click on configure change name in it
then click on done after this either you add new world by clicking on add or
click on continue then you will be able to see new name in dropdown menu 

### Outstanding before merging

If anything. You can use neat checkboxes! Feel free to delete if not needed

- [ ] need to refresh page after clicking on back button in configure this can be done by creating a refresh button below dropdown menu

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java [hunk 1]
```diff
@@ -103,6 +104,8 @@
     private Context context;
     private int worldNumber;
     private String selectedWorld = "";
+    public int camelLocation;
```
Comment (Cervator) [inline engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java]: "camel" ? Huh?

File: engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java [hunk 2]
```diff
@@ -174,6 +177,7 @@ public String get() {
             @Override
             public void set(String value) {
                 selectedWorld = value;
+                camelLocation = findIndex(worlds,selectedWorld);
```
Comment (Cervator) [inline engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java]: Trivial code style issue with whitespace. Please be sure to set up and run Checkstyle on any files you edit to catch those warnings :-)

File: engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java [hunk 3]
```diff
@@ -193,7 +198,8 @@ public void set(String value) {
             } catch (UnresolvedWorldGeneratorException e) {
                 logger.error("Can't configure the world! due to {}", e.getMessage());
             }
-        });
+        }
```
Comment (Cervator) [inline engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java]: No need to split this across two-lines. The one-line approach matches the current style. Might be some sort of auto-formatting in your IDE?

File: engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java [hunk 4]
```diff
@@ -315,6 +330,19 @@ public void setEnvironment(UniverseWrapper wrapper) {
         }
     }
 
+    public int findIndex(List<WorldSetupWrapper> worlds,String worldName)
+    {
+
+        for(int i = 0 ;i<worlds.size();i++) {
```
Comment (Cervator) [inline engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java]: More whitespace issues here and elsewhere. Checkstyle please :-)

File: engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java [hunk 5]
```diff
@@ -315,6 +330,19 @@ public void setEnvironment(UniverseWrapper wrapper) {
...
+            WorldSetupWrapper random = worlds.get(i);
+            Name customName= random.getWorldName();
+            if(customName.toString().equals(worldName)) {
+                return(i);
+            }
+        }
+        return(-1);
```
Comment (Cervator) [inline engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java]: No need for `( )` here

File: engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java [hunk 6]
```diff
@@ -315,6 +330,19 @@ public void setEnvironment(UniverseWrapper wrapper) {
...
     }
 
+    public int findIndex(List<WorldSetupWrapper> worlds,String worldName)
+    {
+
+        for(int i = 0 ;i<worlds.size();i++) {
+            WorldSetupWrapper random = worlds.get(i);
+            Name customName= random.getWorldName();
+            if(customName.toString().equals(worldName)) {
+                return(i);
```
Comment (Cervator) [inline engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java]: Or `( )` here

File: engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java [hunk 7]
```diff
@@ -315,6 +330,19 @@ public void setEnvironment(UniverseWrapper wrapper) {
         }
     }
 
+    public int findIndex(List<WorldSetupWrapper> worlds,String worldName)
```
Comment (Cervator) [inline engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java]: You might want to add Javadoc to a place like this to explain the return value (that `-1` means the index value wasn't found)

File: engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/WorldSetupScreen.java [hunk 8]
```diff
@@ -67,22 +69,36 @@
     private ModuleEnvironment environment;
     private Context context;
     private WorldConfigurator oldWorldConfig;
+    private Name camelCase;
```
Comment (Cervator) [inline engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/WorldSetupScreen.java]: More "huh" at the "camel" here - at least it makes sense to some degree that you're involving camelCase the capitalization pattern, but that doesn't tell the user anything about what this variable is for. Name by its purpose not its formatting.

File: engine/src/main/resources/assets/ui/worldSetupScreen.ui [hunk 9]
```diff
@@ -83,6 +83,30 @@
                   }
                 ]
               },
+              {
+                "type": "RowLayout",
+                "id": "custom_name",
```
Comment (Cervator) [inline engine/src/main/resources/assets/ui/worldSetupScreen.ui]: Generally avoid `_` in naming and use camel case styling instead (so `customName`) - but ideally be more descriptive like `worldName` or `customWorldName`

**Pull request closed without merging**